src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,36 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder result = new StringBuilder("Report for period ");
+        result.append(dateFrom).append(" - ").append(dateTo).append(System.lineSeparator());
+        for (String name : names) {
+            result.append(name).append(" - ");
+            int sum = 0;
+            for (String info : data) {
+                if (info.contains(name) && dateValid(dateFrom, info.substring(0, 10), dateTo)) {
+                    sum += Integer.parseInt(info.split(" ")[2])
+                            * Integer.parseInt(info.split(" ")[3]);
+                }
+            }
+            result.append(sum).append(System.lineSeparator());
+        }
+        return result.toString().trim();
+    }
+
+    public static boolean dateValid(String dateFrom, String date, String dateTo) {
```

Comment:
cool! separate method for validation!
and maybe we can pass here instances of `LocalDate`